elements/Shape.js

```diff
@@ -10,6 +11,17 @@ class Shape extends Component {
         });
         this.state = this.touchableGetInitialState();
     }
+
+    setNativeProps = (args) => {
+        var extracted = extractProps(args, this)
```

Comment:
Correct me if I'm wrong. `extractProps` does not expect matrices. It expects transform objects. As a result, if I pass a matrix directly, it will modify it to the identity matrix. And ofc, the for loop will let the matrix pass. 

Namely, I think the original design is to pass transform objects and the `extractProps ` will translate it into matrix. With your changes, it cannot do this anymore.